### PR TITLE
HDDS-9524. Clean up wait for leader OM in integration tests

### PR DESCRIPTION
**Pull request by david1859168**
## What changes were proposed in this pull request?
MiniOzoneHAClusterImpl:

getOMLeader() returns current OM leader.
getOMLeader(boolean) optionally waits until OM leader is elected
There are some tests that implement the same "wait for leader OM" logic using the non-waiting getOMLeader(). 

```
hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestAddRemoveOzoneManager.java
209:    GenericTestUtils.waitFor(() -> cluster.getOMLeader() != null, 500, 30000);
437:    GenericTestUtils.waitFor(() -> cluster.getOMLeader() != null, 500, 30000);

hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHA.java
466:        getCluster().getOMLeader() != null, 500, timeout);

hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHASnapshot.java
126:    await(120_000, 100, () -> cluster.getOMLeader() != null);
253:    await(180_000, 100, () -> cluster.getOMLeader() != null);
```

Goals of this task:
rename getOMLeader(boolean) to waitForLeaderOM() (the method is only called with true)
make waitForLeaderOM() public
replace duplicated wait logic in tests with a call to waitForLeaderOM()

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-9524

## How was this patch tested?
Made changes and ran unit test.

### CODE REVIEW

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHA.java [hunk 1]
```diff
@@ -462,7 +461,6 @@ protected void waitForLeaderToBeReady()
     // Wait for Leader Election timeout
     int timeout = OZONE_OM_RATIS_SERVER_FAILURE_TIMEOUT_DURATION_DEFAULT
         .toIntExact(TimeUnit.MILLISECONDS);
```
Comment (adoroszlai) [inline hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHA.java:463]: `timeout` is now unused.  Please remove it, as well as the import of `OZONE_OM_RATIS_SERVER_FAILURE_TIMEOUT_DURATION_DEFAULT`.

**Review by adoroszlai — COMMENTED** [review]
Thanks @david1859168 for the patch.  Findbugs reports an unused variable:

```
M D DLS: Dead store to timeout in org.apache.hadoop.ozone.om.TestOzoneManagerHA.waitForLeaderToBeReady()  At TestOzoneManagerHA.java:[line 463]
```

https://github.com/david1859168/ozone/actions/runs/7652102229/job/20851218541#step:7:11

File: hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHA.java (hunk 1 above)
Comment (david1859168) [reply hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestOzoneManagerHA.java]: Fixed it re-commited.

**Review by adoroszlai — APPROVED** [review]
Thanks @david1859168 for updating the patch, LGTM.

**Review by aswinshakil — APPROVED** [review]
Nice find. Thanks for the patch @david1859168 and Thanks for the review @adoroszlai

**Comment by adoroszlai** [comment]
Thanks @david1859168 for the patch, @aswinshakil for the review.

**Pull request merged**